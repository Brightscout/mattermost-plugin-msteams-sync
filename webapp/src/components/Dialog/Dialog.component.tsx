import React from 'react';
import {useDispatch} from 'react-redux';

import {Dialog as MMDialog, LinearProgress, DialogProps} from '@brightscout/mattermost-ui-library';

import usePluginApi from 'hooks/usePluginApi';
import {getDialogState} from 'selectors';
import {closeDialog} from 'reducers/dialog';

export const Dialog = ({onCloseHandler, onSubmitHandler}: Pick<DialogProps, 'onCloseHandler' | 'onSubmitHandler'>) => {
    const dispatch = useDispatch();
    const {state} = usePluginApi();
    const {show, title, description, destructive, primaryButtonText, secondaryButtonText, isLoading} = getDialogState(state);

    const handleClose = () => dispatch(closeDialog());

    return (
        <MMDialog
            description={description}
            destructive={destructive}
            show={show}
            primaryButtonText={primaryButtonText}
            secondaryButtonText={secondaryButtonText}
            onCloseHandler={() => {
                handleClose();
                onCloseHandler();
            }}
            onSubmitHandler={onSubmitHandler}
            className='disconnect-dialog'
            title={title}
        >
            {isLoading && <LinearProgress/>}
        </MMDialog>
    );
};
