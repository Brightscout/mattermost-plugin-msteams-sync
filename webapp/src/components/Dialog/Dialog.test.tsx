import React from 'react';

import {render, RenderResult} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {DialogProps, LinearProgress} from '@brightscout/mattermost-ui-library';

import {Dialog} from './Dialog.component';

const onCloseHandler = jest.fn();
const onSubmitHandler = jest.fn();

const dialogProps = {
    onCloseHandler,
    onSubmitHandler,
};

let tree: RenderResult;

describe('Dialog', () => {
    beforeEach(() => {
        tree = render(<Dialog {...dialogProps}/>);
    });

    it('Should render correctly', () => {
        expect(tree).toMatchSnapshot();
    });

    it('Should close the dialog on clicking close button', () => {
        expect(tree.getAllByRole('button').length).toEqual(2);

        userEvent.click(tree.getAllByRole('button')[1]);
        expect(onCloseHandler).toHaveBeenCalledTimes(1);
    });
});
