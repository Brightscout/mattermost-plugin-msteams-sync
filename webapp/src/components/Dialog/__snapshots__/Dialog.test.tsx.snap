// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    style="padding-right: 0px;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade sc-fFSPTT huYLjr mm-dialog disconnect-dialog modal show"
      role="dialog"
      style="display: block;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="sc-bkbkJK cHKkeq"
          >
            <h2
              class="sc-dIvrsQ iMFmGO"
            >
              mockTitle
            </h2>
            <button
              class="sc-fKgJPI dGklgm"
            >
              <div
                class="sc-bdnxRM dOvjAb mm-icon"
                size="20"
              >
                <reactcomponent />
              </div>
            </button>
          </div>
          <div
            class="sc-iemWCZ lnrLZX"
          >
            
            <div
              class="sc-bCwfaz fFyvgR"
            >
              <div
                class="mm-linearProgress  linear_loader  progress"
              >
                <div
                  aria-valuemax="100"
                  aria-valuemin="0"
                  class="progress-bar"
                  role="progressbar"
                />
              </div>
            </div>
            <div
              class="sc-dlMDgC dEUhfl"
            >
              <button
                class="sc-kEqXSa cwanGm mm-button "
                width="fit-content"
              >
                Close
              </button>
              
            </div>
          </div>
          <div
            class="sc-kfYoZR gFGGdr"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
